internal/api/grpc/server/middleware/activity_interceptor.go

```diff
@@ -0,0 +1,28 @@
+package middleware
+
+import (
+	"context"
+	"strings"
+
+	"google.golang.org/grpc"
+
+	"github.com/zitadel/zitadel/internal/activity"
+)
+
+func ActivityInterceptor() grpc.UnaryServerInterceptor {
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+		resp, err := handler(ctx, req)
+		if isResourceAPI(info.FullMethod) {
+			activity.TriggerWithContext(ctx, activity.ResourceAPI)
+		}
+		return resp, err
+	}
+}
+
+func isResourceAPI(method string) bool {
+	if strings.HasPrefix(method, "/zitadel.management.v1.ManagementService/") ||
+		strings.HasPrefix(method, "/zitadel.admin.v1.AdminService/") {
+		return true
+	}
+	return false
+}
```

Comment:
As you mentioned in the meet that the V2 APIs also need to be added, it would be cleaner to use a slice and utilize the new `slices` package to make a test:

```suggestion
var resourcePrefixes = []string{
	"/zitadel.management.v1.ManagementService/",
	"/zitadel.admin.v1.AdminService/",
}

func isResourceAPI(method string) bool {
	return slices.ContainsFunc(resourcePrefixes, func(prefix string) bool {
		return strings.HasPrefix(method, prefix)
	})
}
```